web/controllers/pair_controller.ex

```diff
@@ -63,14 +57,45 @@ defmodule Pairmotron.PairController do
   end
 
   defp generate_pairs(year, week) do
-    User.active_users
+    users = User.active_users
       |> order_by(:id)
       |> Repo.all
+
+    pair_ids = fetch_pairs(year, week)
+      |> Enum.map(fn(p) -> p.id end)
+
+    query = from up in UserPair,
+            select: up,
+            where: up.pair_id in ^pair_ids,
+            preload: [:pair, :user]
+
+    user_pairs = query
+      |> Repo.all
+
+    determination = PairBuilder.determify(user_pairs, users)
+
+    determination.dead_pairs
+      |> Enum.map(fn(p) -> Repo.delete! p end)
+
+    pairs = determination.remaining_pairs
+      |> Repo.preload([:users, :pair_retros])
+      |> Enum.filter(fn(p) -> length(p.pair_retros) == 0 end)
+
+    results = determination.available_users
       |> Mixer.mixify(week)
-      |> Pairer.generate_pairs
+      |> Pairer.generate_pairs(pairs)
+
+    results.pairs
       |> Enum.map(fn(users) -> make_pairs(users, year, week) end)
       |> List.flatten
       |> Enum.map(fn(p) -> Repo.insert! p end)
+
+    insert_user_pair(results.user_pair)
```

Comment:
I think that this is the perfect place to use `Ecto.Multi` to bring the delete and the inserts into one database transaction to stop us from partially applying changes.